ext/mixed/js/text-scanner.js

```diff
@@ -318,21 +318,26 @@ class TextScanner extends EventDispatcher {
         }
     }
 
-    setTextSourceScanLength(textSource, length) {
+    getTextSourceContent(textSource, length) {
```

Comment:
Is this function intended to leave textSource having the same state as what it entered the function with? If so, probably best to operate on a clone.

Reply:
Yep, that's the idea. I didn't know that clones would have their text selection detached from the page. Thanks! https://github.com/FooSoft/yomichan/pull/500/commits/d6567a3c1d4b3dae0b05ace3b11aad5ceeefba1a